### DISCUSSION TITLE
I deleted a commit, but it is still showing in my activity report on Github?

### DISCUSSION THREAD

**Discussion by Caleb-Cohen** [community/community#30433] [New to GitHub]
<img width="908" alt="Screen Shot 2022-08-22 at 12 20 11 PM" src="https://user-images.githubusercontent.com/75268174/185981306-5094ae7f-51d6-4035-b065-e8909bbb5b37.png">
I accidentally pushed a commit that had a date of 2012 instead of 2022, and have been trying to remove it.

I used `git rebase -i` and changed the commit `No oddities here dated Wed Aug 8 19:48:41 2012` from `pick` to `drop` and it is no longer in my `git log` as seen below. 

![before](https://user-images.githubusercontent.com/75268174/185982615-861eaa49-e5d2-40fc-ae17-0e8aaf82c97a.jpg)
![after](https://user-images.githubusercontent.com/75268174/185982627-e7359bed-e8c6-4f49-90c8-a4503f8ec500.jpg)

Yet after a few hours, I still see activity reported in 2012. Is it possible to delete the activity without being too drastic?

**Comment by radar-blip** [comment]
I think your contribution chart is going to show all activity, including deletions. 
Not sure if you can get rid of that at all. 

Maybe just chalk it up to a mistake.

**Reply by Caleb-Cohen** [reply]
I suspected as much. Thanks for taking the time to weigh in!

**Reply by radar-blip** [reply]
You're welcome!

**Reply by solvaholic** [reply]
> I think your contribution chart is going to show all activity, including deletions.

Asking around some, I've been able to confirm if a commit is deleted from GitHub.com it should _eventually_ disappear from contributions. The process can take time, expect hours or days.

I want to acknowledge: It can be difficult to delete a commit from GitHub. Git will prune any commits that are in the repository and are not reachable, that part's easy. When a commit is reachable, however, making it unreachable is what can be difficult - and destructive. [_Removing sensitive data from a repository_](https://docs.github.com/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository) describes the procedure.

Since we're talking about commits, here, I encourage anyone seeing this to read through [_Commits are snapshots, not diffs_](https://github.blog/2020-12-17-commits-are-snapshots-not-diffs/) - even if you've read it before :+1:

**Comment by solvaholic** [comment]
@Caleb-Cohen In case you'd like to dig further..

> I used `git rebase -i` and changed the commit `No oddities here dated Wed Aug 8 19:48:41 2012` from `pick` to `drop` and it is no longer in my `git log` as seen below.

When `git rebase` dropped commit f8dae7380f, it would have also rewritten commit 5b14d4fb05 with d27504703d as its new parent. Notice the `small enough` commit SHA changed to 2b5f19885d.

If rewriting history this way made commit f8dae7380f unreachable, [Git garbage collection](https://git-scm.com/docs/git-gc) should eventually delete the commit.

Do either of those commits still exist in your local clone? or on GitHub?

You can check locally with commands like these:

```shell
git show f8dae7380f
git show 5b14d4fb05
```

To check on GitHub, you can use URLs like these. Use the correct owner and repository names:

```text
https://github.com/Caleb-Cohen/REPOSITORY/commit/f8dae7380f3c945d5e5fbf79c0346d8692b1da0f
https://github.com/Caleb-Cohen/REPOSITORY/commit/5b14d4fb05e10f63e325a411c415ccdb6e41253d
```

**Reply by Caleb-Cohen** [reply]
`Notice the small enough commit SHA changed to 2b5f19885d.` I did notice that! Also looks like any newer commits, not just `small enough,` had their commit ID changed.

`Do either of those commits still exist in your local clone? or on GitHub?`

Interestingly enough, when I use the information provided it looks like both commits still exist locally and in Github though.

<img width="1464" alt="Screen Shot 2022-08-22 at 4 51 08 PM" src="https://user-images.githubusercontent.com/75268174/186025034-b6ff0dcc-8e9c-4f01-92ee-13b9879ce0f6.png">

**Reply by solvaholic** [reply]
> Interestingly enough, when I use the information provided it looks like both commits still exist locally and in Github though.

Thanks for checking that and following up, @Caleb-Cohen  :+1:

You can use your Git client to get an idea, in your local clone, of why those commits might still be reachable. For example, this will list any branches or tags - seen by your Git client - that contain the commit you wanted to get rid of:

```shell
git for-each-ref --contains f8dae7380f
```

**If it's OK with you to leave this commit as-is,** then I think:

:tada: Congratulations on your second first contribution! :tada:

**If it's important to you that this commit be removed,** please follow these instructions:
<https://docs.github.com/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository>

Pay special attention to the warnings. If you encounter problems, or if you need to delete pull requests, please [raise a support ticket](https://support.github.com/contact) and link to this discussion for reference.

**Reply by Caleb-Cohen** [reply]
Thank you!

I gave the delete method in your link a try and `git for-each-ref --contains f8dae7380f` is now returning `error: malformed object` which I assume is because there is nothing to find. I did try `d275047` and this was the result
<img width="517" alt="Screen Shot 2022-08-22 at 7 33 13 PM" src="https://user-images.githubusercontent.com/75268174/186042977-0fcf9205-5e41-40c9-8dbc-48b585978161.png">

I think at this point I've just accepted I'll have a little green square in 2012, haha. Thank you for taking the time though! I certainly learned a bit about git & github.

**Reply by Amit1706** [reply]
May be it is that day when u had joined the github.
It is also counted as contribution to github.
